src/shop/components/header/CartButton.tsx

```diff
@@ -1,22 +1,45 @@
 import styled from '@emotion/styled';
 import { useAPIDataContext } from '../../../context/APIDataProvider';
 import { getShoppingCartData } from '../../../api/cart';
+import { BasicModal } from '@dev-dino22/modal-components';
+import CartDetails from '../cart-details/CartDetails';
+import { useBoolean } from '../../../hooks/useBoolean';
 
 function CartButton() {
   const { data: cartListData } = useAPIDataContext({
     fetcher: getShoppingCartData,
     name: 'cart',
   });
+  const [isOpenCartModal, openCartModal, closeCartModal] = useBoolean(false);
+
   const itemsCount = cartListData?.length || 0;
   return (
-    <Container data-testid="cart-button">
-      <CartIcon src="./assets/icons/Cart.svg" />
-      {itemsCount > 0 && (
-        <ItemsCountBox>
-          <ItemsCountText data-testid="cart-count">{itemsCount}</ItemsCountText>
-        </ItemsCountBox>
+    <>
+      <Container
+        onClick={openCartModal}
+        role="cart-button"
+        aria-label="장바구니 열기"
+      >
+        <CartIcon src="./assets/icons/Cart.svg" />
+        {itemsCount > 0 && (
+          <ItemsCountBox>
+            <ItemsCountText aria-label="현재 장바구니 목록 수">
+              {itemsCount}
+            </ItemsCountText>
+          </ItemsCountBox>
+        )}
+      </Container>{' '}
```

Comment:
공백 의도한 게 아니라면 없어져도 괜찮을 것 같아 👍
```suggestion
      </Container>
```